addons/mail/static/src/messaging/component/follower_subtype/follower_subtype_tests.js

```diff
@@ -0,0 +1,239 @@
+odoo.define('mail.messaging.component.FollowerSubtypeTests', function (require) {
+'use strict';
+
+const components = {
+    FollowerSubtype: require('mail.messaging.component.FollowerSubtype'),
+};
+const {
+    afterEach: utilsAfterEach,
+    afterNextRender,
+    beforeEach: utilsBeforeEach,
+    pause,
+    start: utilsStart,
+} = require('mail.messaging.testUtils');
+
+QUnit.module('mail', {}, function () {
+QUnit.module('messaging', {}, function () {
+QUnit.module('component', {}, function () {
+QUnit.module('FollowerSubtype', {
+    beforeEach() {
+        utilsBeforeEach(this);
+        this.createFollowerSubtypeComponent = async ({ follower, followerSubtype }) => {
+            const FollowerSubtypeComponent = components.FollowerSubtype;
+            FollowerSubtypeComponent.env = this.env;
+            this.component = new FollowerSubtypeComponent(null, {
+                followerLocalId: follower.localId,
+                followerSubtypeLocalId: followerSubtype.localId,
+            });
+            await this.component.mount(this.widget.el);
+        };
+        this.start = async params => {
+            if (this.widget) {
+                this.widget.destroy();
+            }
+            let { env, widget } = await utilsStart(Object.assign({}, params, {
+                data: this.data,
+            }));
+            this.env = env;
+            this.widget = widget;
+        };
+    },
+    afterEach() {
+        utilsAfterEach(this);
+        if (this.component) {
+            this.component.destroy();
+            this.component = undefined;
+        }
+        if (this.widget) {
+            this.widget.destroy();
+            this.widget = undefined;
+        }
+        this.env = undefined;
+        delete components.FollowerSubtype.env;
```

Comment:
I don't understand exactly but It's made like this in a lot of tests so maybe make an agreement on how to do it and make a commit to apply the potential changes everywhere ?

Reply:
Yes this was more of a general comment that has to be applied everywhere if we do apply it. Fine if you keep the usual stuff here.

Reply:
Yes, let's keep it like this for the moment.
I think we will know how to proceed when components are in `env` (if we do that, still unsure?). For instance, we could imagine components been defined with a factory, so that there are no side-effect whatsoever, and we don't have to set or delete env manually.